pkg/api/router.go

```diff
@@ -14,8 +14,12 @@ import (
 
 func GenerateRouter(apiConf *Config, metrics *Metrics, client *pgclient.Client, elector *util.Elector) http.Handler {
 	router := route.New()
-	writeHandler := timeHandler(metrics.HTTPRequestDuration, "write", Write(client, elector, metrics))
-	router.Post("/write", writeHandler)
+
+	// If we are running in read-only mode, don't handle write route.
+	if !apiConf.ReadOnly {
+		writeHandler := timeHandler(metrics.HTTPRequestDuration, "write", Write(client, elector, metrics))
```

Comment:
Optional: can we replace with a handler that logs the 404 and then returns 404. Would making the logging more useful in that case. But I like this A LOT more.